src/bitcoind.cpp

```diff
@@ -95,8 +95,7 @@ static bool AppInit(int argc, char* argv[])
 
     try
     {
-        if (!fs::is_directory(GetDataDir(false)))
-        {
+        if (!CheckDataDirOption()) {
```

Comment:
In commit "Fix datadir handling in bitcoind" (160470f76f22edd1f9ef6448584381fc5626b9c4)

I can't figure out what this commit is doing. Is it fixing #15745 or #15864? Is the idea to prevent the datadir from being cached here? Can you write in the commit description what problem this is fixing and how it works?

Reply:
Please see https://github.com/bitcoin/bitcoin/pull/15864#issuecomment-490952565